robot/control/Src/motion-control/RobotEstimator.cpp

```diff
@@ -0,0 +1,68 @@
+#include "motion-control/RobotEstimator.hpp"
+#include "rc-fshare/robot_model.hpp"
+#include "MicroPackets.hpp"
+
+extern DebugInfo debugInfo;
+
+RobotEstimator::RobotEstimator(uint32_t dt_us) {
+    const double dt = dt_us/1000000.0;
+
+    // Assume constant velocity
+    F << 1,  0,  0,
+          0, 1,  0,
+          0,  0, 1;
+
+    // Assume no input for the moment
+    // May need to change based on filter lag
+    B << 0, 0, 0, 0,
+         0, 0, 0, 0,
+         0, 0, 0, 0;
+
+    //H = [bot2Wheel;
+    //      0, 0, 1]
+    H.block<4, 3>(0, 0) = RobotModel::get().BotToWheel;
+    H.block<1, 3>(4, 0) << 0, 0, 0;
+
+    Q.setIdentity();
+    Q *= 3.0*processNoise / (dt*dt);
+
+    R.setIdentity();
+    R.block<4, 4>(0, 0) *= encoderNoise;
+    R.block<1, 1>(4, 4) *= gyroNoise;
+
+    P.setIdentity();
+    P *= initCovariance;
+
+    I.setIdentity();
+
+    // Assume no motion
+    x_hat << 0, 0, 0;
+}
+
+// 1.3 ms
```

Comment:
Wait, what? Shouldn't the estimator be running at 200Hz with the rest of the MotionControl stack?

Reply:
We were trying to figure out what was taking so long in this. Iirc we didn't have optimizations on so these functions actually took. It's fixed now

Reply:
Yikes. Okay. As a side note we should really probably be using `float` instead of `double` everywhere, 64-bit floating point is probably overkill for what we're doing and it's almost certainly significantly slower considering we have native hardfp32 support now. Thoughts?

Reply:
The entire controller / estimator is under 300 us now. We could switch if want to based on hardware support, but it's starting to get to the point it doesn't matter. IO is still the bulk of the transactions.
I'm curious to see just how much faster it is if you swapped to that.

Reply:
That 300us is just computations and nothing else, right? Seems a little bit high for what effectively comes down to a few matrix multiplications and an inverse.

Reply:
Scratch that, just checked, it's 133 us (all the module hpps have the estimates), which at a 200 MHz doesn't actually feel that far off. Try it and see

Reply:
Ah. That seems fairly reasonable, then - still a little higher than I'd expect but I'll look into it later if it ends up being a concern.